browser/ui/views/sidebar/sidebar_items_contents_view.cc

```diff
@@ -110,6 +110,28 @@ gfx::Size SidebarItemsContentsView::CalculatePreferredSize() const {
   return views::View::CalculatePreferredSize();
 }
 
+void SidebarItemsContentsView::OnThemeChanged() {
+  View::OnThemeChanged();
+
+  // Refresh favicons for web type items when theme changes.
+  const auto& items = sidebar_model_->GetAllSidebarItems();
+  const size_t items_num = items.size();
+
+  if (items_num != children().size()) {
+    return;
+  }
```

Comment:
Q. I'm curious what's intended here. Does this to imply that OnThemeChange() can be called before item views are attached? 

---

Optional: Comparing `children()` with `items_num` seems to be a little easy to be broken, as it assumes that all children of the contents view are the item view. I mean, it can be easily broken with we need another type of view as a child of contents view but forget to update this. Can we just remove it?